crates/floresta-wire/src/p2p_wire/node.rs

```diff
@@ -183,21 +183,21 @@ pub struct NodeCommon<Chain: BlockchainInterface + UpdatableChainstate> {
     pub(crate) network: Network,
 }
 
-pub struct UtreexoNode<Context, Chain: BlockchainInterface + UpdatableChainstate>(
-    pub(crate) NodeCommon<Chain>,
-    pub(crate) Context,
-);
+pub struct UtreexoNode<Chain: BlockchainInterface + UpdatableChainstate, NodeModule> {
+    pub(crate) common: NodeCommon<Chain>,
+    pub(crate) context: NodeModule,
+}
 
-impl<Chain: BlockchainInterface + UpdatableChainstate, T> Deref for UtreexoNode<T, Chain> {
+impl<Chain: BlockchainInterface + UpdatableChainstate, T> Deref for UtreexoNode<Chain, T> {
```

Comment:
In a next PR, we should get rid of this `Deref` and `DerefMut`. This makes things harder to read if you're not familiar with the code.

Reply:
Yeah, if you don't know about this deref impl you may be confused when we access fields that do not appear in the `UtreexoNode` declaration. It's pretty nice to use the `common` and `context` fields!